pkg/gardenlet/controller/shoot/shoot_control_reconcile.go

```diff
@@ -319,12 +343,17 @@ func (c *Controller) runReconcileShootFlow(o *operation.Operation) *gardencorev1
 		})
 		_ = g.Add(flow.Task{
 			Name:         "Destroying External DNS Entry", // delete DNS entries during hibernation.
-			Fn:           flow.TaskFn(botanist.Shoot.Components.DNS.ExternalEntry.Destroy).DoIf(o.Shoot.HibernationEnabled),
+			Fn:           flow.TaskFn(botanist.Shoot.Components.DNS.ExternalEntry.Destroy).DoIf(o.Shoot.HibernationEnabled).SkipIf(useSNI),
```

Comment:
Why is the deletion of the DNS entries skipped if SNI is used?

Reply:
Because with the previous setup the loadbalancer of the Shoot cluster was removed once the cluster was hibernated. In this case it's not.

Reply:
Hm okay, so what happens if I'm trying to contact the DNS name of a hibernated shoot now with SNI? Isn't it "better" to still delete the DNS entries to prevent requests from hitting the seed-wide load balancer?

Reply:
```
curl -k \
    -HHost:test.foo \
    --resolve test.foo:443:<Seed-LB-IP> \
   https://test.foo:443



* Added test.foo:443:<Seed-LB-IP> to DNS cache
* Hostname test.foo was found in DNS cache
*   Trying <Seed-LB-IP>...
* TCP_NODELAY set
* Connected to test.foo (<Seed-LB-IP>) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/cert.pem
  CApath: none
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* LibreSSL SSL_connect: SSL_ERROR_SYSCALL in connection to test.foo:443
* Closing connection 0
curl: (35) LibreSSL SSL_connect: SSL_ERROR_SYSCALL in connection to test.foo:443
```

Reply:
Isn't it "better" to still delete the DNS entries to prevent requests from hitting the seed-wide load balancer?

Reply:
Removed in latest commit